Automattic/themes
# Fix reveal background

#### Changes proposed in this Pull Request:

It fixes the Hover Reveal style of the Cover block for the latest versions of WordPress.

#### Known issues:

- https://github.com/Automattic/themes/issues/8386
- https://github.com/Automattic/themes/issues/8395

#### Test instructions:

I tested upgrading to all major releases from 5.0 (5.1, 5.2, 5.3...). And I tried to simplify it getting some specific versions. But feel free to test bumping 1 in 1 if you have time. :)

For every update, make sure you update the blocks in the editor by editing anything in the page and clicking on "Update" to get the correct version of the blocks.

In some updates it will also ask for a database update. Confirm it when requested.

- Setup an environment with WP 5.0 (first version with Gutenberg).
  - Create a page with some cover blocks with some variations:
    - With image and background.
    - For each (image and background), use the different options (using the default and changing the color overlay and opacity).
  - Set the style "Hover Reveal" to the blocks.
  - Check that it works properly when you hover the elements.
- Update your WP to 5.2.
  - Update the blocks
  - Make sure everything continues working well.
- Update your WP to 5.3.
  - Create a cover block only with color (without image or video) and the Hover Reveal style.
  - Make sure everything continues working well.
- Update your WP to 5.4.
  - Add variations with gradient, using video, image, and only text.
  - Use a different opacity in one of them.
  - Make sure everything continues working well.
- Update your WP to 5.5.
  - Update the blocks.
  - See that everything continues working, but you will see an issue when hovering the text in the editor. I created [an issue](https://github.com/Automattic/themes/issues/8386) for it.
- Upgrade to WP 5.9, and see that the background will always be visible, regardless if we are in this branch or trunk. I tried to fix it, but it was complex given all possible scenarios and I was also afraid to break for other versions, so I just created [an issue](https://github.com/Automattic/themes/issues/8395) for now since the more important would be to fix it for the latest versions.
- Update to WP 6.0.
  - Make sure everything continues working well.
- Update to WP 6.3.
  - Make sure everything continues working well.
- Update to WP 6.6.
  - Make sure everything continues working well.

https://github.com/user-attachments/assets/ee7d50f8-6775-4776-a242-8230439e2607

#### Related issue(s):

https://github.com/Automattic/themes/issues/8377
